### PR TITLE
Support Database Data Classification

### PR DESCRIPTION
**Pull request by bashahee**
## Description

This PR adds support of database Data Classification via PowerShell, as discussed in the design review https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/241

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Sql/Sql/DataClassification/Cmdlet/GetAzSqlDatabaseSensitivityClassification.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
+﻿using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
```
Comment (cormacpayne) [inline src/Sql/Sql/DataClassification/Cmdlet/GetAzSqlDatabaseSensitivityClassification.cs]: @bashahee quick comment: any file that gets shipped with your module should contain a [Microsoft license header](https://github.com/Azure/azure-powershell/blob/master/src/Sql/Sql/Common/SqlSkuUtils.cs#L1-L13) -- please add this to all of the files you plan on shipping that are missing this

**Comment by vladimir-shcherbakov** [comment]
@bashahee Your ci-build is failing.

**Comment by vladimir-shcherbakov** [comment]
Is it possible for you to [squash](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md#squashing) the commit?

File: src/Sql/Sql/help/Get-AzSqlDatabaseSensitivityClassification.md [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+## OUTPUTS
+
+### Microsoft.Azure.Commands.Sql.DataClassification.Model.SqlDatabaseSensitivityClassificationModel
+
+## NOTES
+
+## RELATED LINKS
+
+Learn more about [Azure SQL Database data discovery and classification](https://docs.microsoft.com/en-us/azure/sql-database/sql-database-data-discovery-and-classification)
```
Comment (cormacpayne) [inline src/Sql/Sql/help/Get-AzSqlDatabaseSensitivityClassification.md]: @bashahee the help generation module, `platyPS`, is expecting the lines provided in this section follow a specific format of strictly hyperlink text and nothing else. I believe the build failures are resolved if you more the "Learn more about" text into the brackets with the "Azure SQL Database data discovery and classification" text. You do can this, or you can move the entire line into the `## NOTES` section. This should be done for every markdown file where you add this "Learn more about" link. Please let us know if you have any questions.

**Comment by bashahee** [comment]
@cormacpayne Can you please approve?

**Comment by cormacpayne** [comment]
@vladimir-shcherbakov assigning back to you since you did the design review for these changes

File: src/Sql/Sql.Test/ScenarioTests/DataClassificationTests.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
+﻿using Microsoft.Azure.Commands.ScenarioTest.SqlTests;
+using Microsoft.WindowsAzure.Commands.ScenarioTest;
```
Comment (vladimir-shcherbakov) [inline src/Sql/Sql.Test/ScenarioTests/DataClassificationTests.cs]: Please add Microsoft licence header

**Comment by vladimir-shcherbakov** [comment]
@bashahee One small comment on the [cmdlet noun  singularity](https://github.com/Azure/azure-powershell/blob/e7943e188a001d189aa304bf6e6c660b4817d070/documentation/development-docs/design-guidelines/cmdlet-best-practices.md#specific-noun-and-noun-singularity):
>To enhance the user experience, the noun that you choose for a cmdlet name should be singular.

For the `Get-AzSqlDatabaseSensitivityRecommendations` & `Get-AzSqlInstanceDatabaseSensitivityRecommendations` cmdlets - is it possible for you to use  `Recommendation` singular noun instead of  `Recommendations` to follow this rule?

**Comment by bashahee** [comment]
> @bashahee One small comment on the cmdlet noun  singularity:
> 
> To enhance the user experience, the noun that you choose for a cmdlet name should be singular.
> 
> For the Get-AzSqlDatabaseSensitivityRecommendations & Get-AzSqlInstanceDatabaseSensitivityRecommendations cmdlets - is it possible for you to use  Recommendation singular noun instead of  Recommendations to follow this rule?

No, there is no option to retrieve only one recommendation, i.e. we do not support retrieving a recommendation of a single column. Recommendations are computed on a database.

**Comment by vladimir-shcherbakov** [comment]
@bashahee Let me quote another paragraph from the [Cmdlet Best Practices](https://github.com/Azure/azure-powershell/blob/e7943e188a001d189aa304bf6e6c660b4817d070/documentation/development-docs/design-guidelines/cmdlet-best-practices.md#cmdlet-best-practices) doc:

> To enhance the user experience, the noun that you choose for a cmdlet name should be singular. For example, use the name Get-Process instead of Get-Processes. It is best to follow this rule for all cmdlet names, even when a cmdlet is likely to act upon more than one item.

Please confirm that the rule above doesn't work in your case .

File: src/Sql/Sql.Test/ScenarioTests/DataClassificationTests.ps1 [hunk 4]
```diff
@@ -0,0 +1,530 @@
...
+Gets the values of the parameters used at the tests
+#>
+function Get-DataClassificationTestEnvironmentParameters ($testSuffix)
+{
+	return @{ rgname = "dc-cmdlet-test-rg" +$testSuffix;
+			  serverName = "dc-cmdlet-server" +$testSuffix;
+			  databaseName = "dc-cmdlet-db" + $testSuffix;
+			  loginName = "testlogin";
+			  pwd = "testp@ssMakingIt1007Longer";
+		}
```
Comment (vladimir-shcherbakov) [inline src/Sql/Sql.Test/ScenarioTests/DataClassificationTests.ps1]: Was the line an issue with Credential Scanner?

Comment (bashahee) [reply src/Sql/Sql.Test/ScenarioTests/DataClassificationTests.ps1]: No.

**Comment by bashahee** [comment]
@vladimir-shcherbakov Relevant cmdlets were renamed, following your request.

**Comment by vladimir-shcherbakov** [comment]
singing: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/psnetsign-esrp/259/

**Pull request merged**